_posts/2020-3-27-SciML.md

```diff
@@ -0,0 +1,392 @@
+
+### We provide high-level implementations of the latest algorithms in scientific machine learning
+
+The translational step of bringing new methods of computational science to
+scientists in application areas is what will allow next-generation exploration to occur. We
+provide libraries like:
+
+- [DiffEqFlux.jl](https://github.com/SciML/DiffEqFlux.jl) for neural and universal differential equations
+- [DataDrivenDiffEq.jl](https://github.com/SciML/DataDrivenDiffEq.jl) for automated equation generation with Dynamic Mode Decomposition (DMD) and SInDy type methods
+- [ReservoirComputing.jl](https://github.com/SciML/ReservoirComputing.jl) for echo state networks and prediction of chaotic systems
+- [NeuralNetDiffEq.jl](https://github.com/SciML/NeuralNetDiffEq.jl) for Physics-Informed Neural Networks (PINNs) and Deep BSDE solvers of 100 dimensional PDEs
+
+We will continue to expand this portion of our offering, building tools that
+automatically solve PDEs from a symbolic description using neural networks,
+and generate mesh-free discretizers.
+
+### We provide users of all common scientific programming languages the ability to use our tooling
+
+While the main source of our tooling is centralized in the [Julia programming language](https://julialang.org/),
+we see Julia as a "language of libraries", like C++ or Fortran, for developing
+scientific libraries that can be widely used across the whole community. We
+have previously demonstrated this capability with tools like [diffeqpy](https://github.com/SciML/diffeqpy)
+and [diffeqr](https://cran.r-project.org/web/packages/diffeqr/index.html) for
+using DifferentialEquations.jl from Python and R respectively, and we plan to
+continue along these lines to allow as much of our tooling as possible be accessible
+from as many languages as possible. While there will always be some optimizations
+that can only occur when used from the Julia programming language, DSL builders
+like [ModelingToolkit.jl](https://github.com/SciML/ModelingToolkit.jl) will be
+used to further expand the capabilities and performance of our wrappers.
+
+Here's an example which solves stochastic differential equations with high order adaptive methods
+
+from Python:
+
+```py
+# pip install diffeqpy
+from diffeqpy import de
+# diffeqpy.install()
+
+def f(du,u,p,t):
+    x, y, z = u
+    sigma, rho, beta = p
+    du[0] = sigma * (y - x)
+    du[1] = x * (rho - z) - y
+    du[2] = x * y - beta * z
+
+def g(du,u,p,t):
+    du[0] = 0.3*u[0]
+    du[1] = 0.3*u[1]
+    du[2] = 0.3*u[2]
+
+numba_f = numba.jit(f)
+numba_g = numba.jit(g)
+u0 = [1.0,0.0,0.0]
+tspan = (0., 100.)
+p = [10.0,28.0,2.66]
+prob = de.SDEProblem(numba_f, numba_g, u0, tspan, p)
+sol = de.solve(prob)
+
+# Now let's draw a phase plot
+
+ut = numpy.transpose(sol.u)
+from mpl_toolkits.mplot3d import Axes3D
+fig = plt.figure()
+ax = fig.add_subplot(111, projection='3d')
+ax.plot(ut[0,:],ut[1,:],ut[2,:])
+plt.show()
+```
+

```

Comment:
Maybe mention efforts / plans around static compilation in order to deliver julia interop to other languages with less friction.